keras_nlp/samplers/sampler.py

```diff
@@ -172,18 +172,6 @@ def _validate_prompt_and_mask(self, prompt, mask):
         prompt = tf.ragged.boolean_mask(prompt, mask)
         return prompt, mask
 
-    def _validate_token_probability_fn(
```

Comment:
I think we need to ditch this. It just isn't going to be performant enough. Especially right now outside of the compiled function.